common.c

```diff
@@ -107,7 +107,7 @@ char* create_temp_file(const char* prefix, const char* content) {
     if (-1 == asprintf(&fname, "/tmp/%sXXXXXXXXXX", prefix)) {
         return NULL;
     }
-    if (!mktemp(fname)) {
+    if (!mkstemp(fname)) {
```

Comment:
mkstemp() in contrast to mktemp() opens file, so it needs to be closed. Further in this function, the file is opened again to write the content to it. If we use mkstemp(), we should operate with the descriptor it returned instead closing and opening the file again.